packages/@tinacms/toolkit/src/plugins/branch-switcher/BranchSwitcher.tsx

```diff
@@ -130,16 +131,29 @@ export const BranchSwitcher = ({
             x?.indexStatus?.status === 'unknown'
         )
         .forEach(async (x) => {
-          const indexStatus = await cms.api.tina.waitForIndexStatus({
+          const [prom, cancel] = cms.api.tina.waitForIndexStatus({
             ref: x.name,
           })
-          setBranchList((prev) => {
-            const newList = Array.from(prev)
-            const index = newList.findIndex((y) => y.name === x.name)
-            newList[index].indexStatus = indexStatus
-            return newList
-          })
+          cancelFuncs.push(cancel)
+          prom
+            .then((indexStatus) => {
+              setBranchList((prev) => {
+                const newList = Array.from(prev)
+                const index = newList.findIndex((y) => y.name === x.name)
+                newList[index].indexStatus = indexStatus
+                return newList
+              })
+            })
+            .catch((e) => {
+              if (e.message === 'AsyncPoller: cancelled') return
+              console.error(e)
```

Comment:
looks like we return before this line

Reply:
Yes that is on purpose. If we get that error back there is no reason to log out out.